lotman/lotman.go

```diff
@@ -0,0 +1,655 @@
+					},
+				},
+				MPA: &MPA{
+					DedicatedGB:     &initDedicatedGB,
+					OpportunisticGB: &initOpportunisticGB,
+					MaxNumObjects:   &Int64FromFloat{Value: 0},
+					CreationTime:    &Int64FromFloat{Value: 0},
+					ExpirationTime:  &Int64FromFloat{Value: 0},
+					DeletionTime:    &Int64FromFloat{Value: 0},
+				},
+			}
+
+			log.Debugf("Creating the root lot defined by %v", rootLot)
+			lotJSON, err := json.Marshal(rootLot)
+			if err != nil {
+				log.Errorf("Error marshalling root lot JSON: %v", err)
+				return false
+			}
+
+			ret = LotmanAddLot(string(lotJSON), &errMsg)
+			if ret != 0 {
+				trimBuf(&errMsg)
+				log.Errorf("Error creating root lot: %s", string(errMsg))
+				return false
+			}
+		}
+		log.Infof("Created root lot")
+	}
+
+	// Now instantiate any other lots that are in the config
+	for _, lot := range initializedLots {
+		if lot.LotName != "default" && lot.LotName != "root" {
+			lotJSON, err := json.Marshal(lot)
+			if err != nil {
+				log.Errorf("Error marshalling lot JSON for %s: %v", lot.LotName, err)
+				return false
+			}
+
+			ret = LotmanAddLot(string(lotJSON), &errMsg)
+			if ret != 0 {
+				trimBuf(&errMsg)
+				log.Errorf("Error creating lot %s: %s", lot.LotName, string(errMsg))
+				return false
+			}
+		}
+	}
+
+	log.Infof("LotMan initialization complete")
+	lotmanInitSuccess = true
+	return true
+}
+
+func CreateLot(newLot *Lot, caller string) error {
+	success := InitLotman()
+	if !success {
+		return errors.New("Failed to initialize LotMan")
+	}
+
+	// Marshal the JSON into a string for the C function
+	lotJSON, err := json.Marshal(*newLot)
+	if err != nil {
+		return errors.Wrapf(err, "Error marshalling lot JSON: %v", err)
+	}
+
+	// Set the context to the incoming lot's owner:
+	errMsg := make([]byte, 2048)
+	callerMutex.Lock()
+	defer callerMutex.Unlock()
+	ret := LotmanSetContextStr("caller", caller, &errMsg)
+	if ret != 0 {
+		trimBuf(&errMsg)
+		return fmt.Errorf(fmt.Sprintf("Error creating lot: %s", string(errMsg)))
+	}
+
+	// Now finally add the lot
+	ret = LotmanAddLot(string(lotJSON), &errMsg)
+	if ret != 0 {
+		trimBuf(&errMsg)
+		return fmt.Errorf(fmt.Sprintf("Error creating lot: %s", string(errMsg)))
+	}
+
+	return nil
+}
+
+func GetLot(lotName string, recursive bool) (*Lot, error) {
+	success := InitLotman()
+	if !success {
+		return nil, errors.New("Failed to initialize LotMan")
+	}
+
+	// Haven't given much thought to these buff sizes yet
+	outputBuf := make([]byte, 4096)
+	errMsg := make([]byte, 2048)
+
+	ret := LotmanGetLotJSON(lotName, recursive, &outputBuf, &errMsg)
+	if ret != 0 {
+		trimBuf(&errMsg)
+		return nil, errors.Errorf("Error getting lot JSON: %s", string(errMsg))
+	}
+	trimBuf(&outputBuf)
+	var lot Lot
+	err := json.Unmarshal(outputBuf, &lot)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error unmarshalling lot JSON: %v", err)
+	}
+	return &lot, nil
+}
+
+func UpdateLot(lotUpdate *LotUpdate, caller string) error {
+	success := InitLotman()
```

Comment:
Can you do a header for this? What does updating a lot entail? What specifically is being updated?